src/lib.rs

```diff
@@ -444,18 +445,19 @@ fn process_join(from: &Arc<Session>, room_id: RoomId, user_id: UserId, subscribe
     }
 
     let mut switchboard = SWITCHBOARD.write()?;
-    let body = json!({ "users": { room_id.as_str(): switchboard.get_users(&room_id) }});
+    let room_users = switchboard.get_room_users(&room_id).collect::<HashSet<_>>();
+    let body = json!({ "users": { room_id.as_str(): room_users }});
 
     // hack -- use data channel subscription to infer this, it would probably be nicer if
     // connections announced explicitly whether they were a publisher or subscriber
     let gets_data_channel = subscribe.as_ref().map(|s| s.data).unwrap_or(false);
     let join_kind = if gets_data_channel { JoinKind::Publisher } else { JoinKind::Subscriber };
 
     if join_kind == JoinKind::Publisher {
-        if switchboard.publishers_occupying(&room_id).len() > config.max_room_size {
+        if room_users.len() > config.max_room_size {
```

Comment:
Previously we iterated twice over the `publishers_by_room` HashMap because `get_users` called `publishers_occupying` too. So this is nice we are only doing it once now.